medical-portal/src/API/RSBC.DMF.MedicalPortal.API/AutoMapper.cs

```diff
@@ -25,13 +29,43 @@ public MappingProfile()
 
             CreateMap<LegacyDocument, ViewModels.CaseDocument>()
              .ForMember(dest => dest.DueDate, opt => opt.MapFrom(src => src.DueDate))
-             .ForMember(dest => dest.CreatedOn, opt => opt.MapFrom(src => src.CreateDate));
+             .ForMember(dest => dest.CreatedOn, opt => opt.MapFrom(src => src.CreateDate))
+             .ForMember(dest => dest.SubmittalStatus, opt => opt.MapFrom(src => GroupSubmittalStatus(src.SubmittalStatus)));
 
             CreateMap<DocumentSubType, ViewModels.DocumentSubTypes>();
 
             CreateMap<EndorsementDto, Endorsement>();
             CreateMap<PidpAdapter.Licence, ViewModels.Licence>();
         }
+
+        private string GroupSubmittalStatus(string submittalStatus)
```

Comment:
instead of duplicating a method, move it to SharedUtils. You can install Enums.NET to SharedUtils, it's lightweight and just remove the logError

Reply:
`    public static class SubmittalStatusExtensions
    {
        public static string Group(this string submittalStatus)
        {
            var canParseEnum = Enums.TryParse<SubmittalStatus>(submittalStatus, true, out var submittalStatusEnum, EnumFormat.Description);
            if (!canParseEnum)
                return submittalStatus;

            switch (submittalStatusEnum)
            {
                case SubmittalStatus.Noncomply:
                case SubmittalStatus.ActionedNoncomply:
                    return SubmittalStatus.Noncomply.AsString(EnumFormat.Description);
                case SubmittalStatus.Received:
                case SubmittalStatus.CleanPass:
                case SubmittalStatus.ManualPass:
                case SubmittalStatus.Reviewed:
                case SubmittalStatus.UnderReview:
                    return SubmittalStatus.Received.AsString(EnumFormat.Description);
                case SubmittalStatus.Rejected:
                    return SubmittalStatus.Rejected.AsString(EnumFormat.Description);
                case SubmittalStatus.Uploaded:
                    return SubmittalStatus.Uploaded.AsString(EnumFormat.Description);
                default:
                    return submittalStatus;
            }
        }
    }`

Reply:
I have moved this method to shared Utils